### ISSUE TITLE
Initial Block Processing Logic

### ISSUE THREAD

**Issue by jrhea** [Consensys-Incorporated/teku#137]
### Description
Implement the block processing logic as described in this section of the [beacon chain spec](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#per-block-processing)

> Note: Not all the logic in the spec is finalized.  Follow Danny Ryan's recommendation by implementing the logic that is well defined.  See pseudo code below.

***pseudo code:***

```
    # per-block processing
    if block:
        # handle block header
        verify_signature(state, block)
        verify_and_update_randao(state, block)
        
        # handle block body operations
        process_attestations(state, block)
```

### Acceptance Criteria
* Add the logic to the blockProcessor method in StateTransition.java - see [here](https://github.com/PegaSysEng/artemis/blob/bdae8d870851d15f8a7f46c2ff4a67741398aab7/artemis/src/main/java/tech/pegasys/artemis/services/beaconchain/StateTransition.java#L62)
* Add appropriate tests

**Comment by jrhea** [comment]
Blocked by #145

**Comment by jrhea** [comment]
closed by #230 and #232
